elastic/e2e-testing
# chore: use DEVELOPER_MODE to not uninstall/unenroll the agent

- chore: do not uninstall/unenroll the agent in DEV_MODE
- fix: pass local DEVELOPER_MODE env to the remote instance



## What does this PR do?
It guards the code block that uninstalls/unenrolls the agent **after** each test scenario with the value of the DEVELOPER_MODE environment variable.

To allow so, we need to pass the local state of that variable to the remote instance using Ansible.



## Why is it important?
We'd like to provide a way to simply run one scenario from local, and inspect the state of the agent right after the scenario: logs, file structure, etc.

Therefore, a developer would be able to setup a test node, run the testss, and check the agent.



## Checklist







## How to test this


0. Create a stack node: `make -C .ci create-stack`
1. Setup local environment for running on Debian 11: `make -C .ci set-env-debian_11_amd64`
2. Source environment: `source ./.ci/.node-debian_11_amd64-env`
3. Create a test node: `make -C .ci create-node`
4. Run one single test in DEVELOPER_MODE: `make -C .ci run-tests DEVELOPER_MODE=true TAGS="fleet_mode && install"`
5. Ansible should present this message:

```
TASK [Extend environment for Fleet testing] ******************************************************************************************************************************************************************************************************************
changed: [3.145.6.222] => (item=SUITE="fleet")
changed: [3.145.6.222] => (item=PROVIDER="remote")
changed: [3.145.6.222] => (item=ELASTICSEARCH_PASSWORD="changeme")
changed: [3.145.6.222] => (item=KIBANA_PASSWORD="changeme")
changed: [3.145.6.222] => (item=SKIP_PULL="1")
changed: [3.145.6.222] => (item=DEVELOPER_MODE="true")
```

6. And the Go execution logs should include:

```
time="2022-06-28T16:13:57Z" level=info msg="Running in Developer mode 💻: runtime dependencies between different test runs will be reused to speed up dev cycle"
```

7. When SSH'ing into the node with `make -C .ci ssh-node`, the agent should be running:

```
ps aux | grep agent
root       17529 10.7  0.3 1567152 56572 ?       Ssl  16:14   0:08 elastic-agent
root       17547  0.0  0.0      0     0 ?        Zs   16:14   0:00 [elastic-agent] <defunct>
root       17570  0.0  0.0      0     0 ?        Zs   16:14   0:00 [elastic-agent] <defunct>
root       17588  1.1  0.8 1525568 137896 ?      Sl   16:14   0:00 /opt/Elastic/Agent/data/elastic-agent-ec48f6/install/metricbeat-8.4.0-SNAPSHOT-linux-x86_64/metricbeat -E setup.ilm.enabled=false -E setup.template.enabled=false -E management.enabled=true -E logging.level=debug -E gc_percent=${METRICBEAT_GOGC:100} -E metricbeat.config.modules.enabled=false -E logging.level=info -E http.enabled=true -E http.host=unix:///opt/Elastic/Agent/data/tmp/default/metricbeat/metricbeat.sock -E logging.files.path=/opt/Elastic/Agent/data/elastic-agent-ec48f6/logs/default -E logging.files.name=metricbeat -E logging.files.keepfiles=7 -E logging.files.permission=0640 -E logging.files.interval=1h -E path.data=/opt/Elastic/Agent/data/elastic-agent-ec48f6/run/default/metricbeat--8.4.0-SNAPSHOT
root       17606  1.3  0.8 1483524 134588 ?      Sl   16:14   0:00 /opt/Elastic/Agent/data/elastic-agent-ec48f6/install/filebeat-8.4.0-SNAPSHOT-linux-x86_64/filebeat -E setup.ilm.enabled=false -E setup.template.enabled=false -E management.enabled=true -E logging.level=debug -E gc_percent=${FILEBEAT_GOGC:100} -E filebeat.config.modules.enabled=false -E logging.level=info -E http.enabled=true -E http.host=unix:///opt/Elastic/Agent/data/tmp/default/filebeat/filebeat.sock -E logging.files.path=/opt/Elastic/Agent/data/elastic-agent-ec48f6/logs/default -E logging.files.name=filebeat -E logging.files.keepfiles=7 -E logging.files.permission=0640 -E logging.files.interval=1h -E path.data=/opt/Elastic/Agent/data/elastic-agent-ec48f6/run/default/filebeat--8.4.0-SNAPSHOT
root       17622  0.5  0.6 1409536 102104 ?      Sl   16:14   0:00 /opt/Elastic/Agent/data/elastic-agent-ec48f6/install/filebeat-8.4.0-SNAPSHOT-linux-x86_64/filebeat -E setup.ilm.enabled=false -E setup.template.enabled=false -E management.enabled=true -E logging.level=debug -E gc_percent=${FILEBEAT_GOGC:100} -E filebeat.config.modules.enabled=false -E logging.level=info -E path.data=/opt/Elastic/Agent/data/elastic-agent-ec48f6/run/default/filebeat--8.4.0-SNAPSHOT--36643631373035623733363936343635
root       17640  0.6  0.8 1451836 130816 ?      Sl   16:14   0:00 /opt/Elastic/Agent/data/elastic-agent-ec48f6/install/metricbeat-8.4.0-SNAPSHOT-linux-x86_64/metricbeat -E setup.ilm.enabled=false -E setup.template.enabled=false -E management.enabled=true -E logging.level=debug -E gc_percent=${METRICBEAT_GOGC:100} -E metricbeat.config.modules.enabled=false -E logging.level=info -E path.data=/opt/Elastic/Agent/data/elastic-agent-ec48f6/run/default/metricbeat--8.4.0-SNAPSHOT--36643631373035623733363936343635
admin      17691  0.0  0.0   3180   652 pts/0    R+   16:15   0:00 grep agent
```

## Related issues


- Closes #2712
